### ISSUE TITLE
Function definition with type variables not recognized as a function

### ISSUE THREAD

**Issue by jfehrle** [reasonml-editor/reasonml-idea-plugin#270]
plugin version: 0.92-2020.2

Note the "V" in the structure panel.

`type s tr r.` defines 3 type variables that are used in the following list of arguments.

![image](https://user-images.githubusercontent.com/1253341/93659118-ccb66d00-f9f6-11ea-9fe0-cd854be753cf.png)

let rec parser_of_tree : type s tr r. s ty_entry -> int -> int -> (s, tr, r) ty_tree -> r parser_t =
  ()

**Comment by giraud** [comment]
I don't understand the syntax nor the expectation

**Comment by jfehrle** [comment]
This example is more correct (though still lacking some of the referenced types):

```
let rec parser_of_tree : type s tr r. s ty_entry -> int -> int -> (s, tr, r) ty_tree -> r parser_t =
  fun entry nlevn alevn -> ()
```

Defines the function `parser_of_tree` while explicitly defining the type of the function.  The formal parameters are `entry`, `nlevn` and `alevn`.  The expectation is that it appears as a function in the structure panel and shows the type there.

The following function definitions are exactly equivalent.  The latter gives the type explicitly rather than having the compiler infer its type:

```
let rec mem x = function
    [] -> false
  | a::l -> compare a x = 0 || mem x l

let rec mem : type a.    a -> a list -> bool    = fun x -> function
    [] -> false
  | a::l -> compare a x = 0 || mem x l
```

The type notation is similar to what you see in a `.mli` file (`val mem : 'a -> 'a list -> bool`).
`a list` is a polymorphic type: it's a list containing items of type 'a'.  The function checks whether an item belongs to a list.

Does that make sense?  Any further questions?

**Comment by giraud** [comment]
Thanks, it's clear

**Comment by giraud** [comment]
fixed in 0.93

**Comment by jfehrle** [comment]
Looks good, thanks.
